server/graphql/server.js

```diff
@@ -0,0 +1,30 @@
+import { ApolloServer } from "apollo-server-express";
+import jwt from "jsonwebtoken";
+
+import { schema } from "./schema";
+
+function getUser(token) {
+  const user = { loggedIn: true };
+  try {
+    jwt.verify(token, "secret");
+  } catch (err) {
+    user.loggedIn = false;
+  }
+  return user;
+}
+
+const publicOperations = ["login"];
+
+export function createApolloServer() {
+  const context = ({ req }) => {
+    const contextObj = {};
+    const token = req.headers.authorization || "";
+    const user = getUser(token);
+    if (!publicOperations.includes(req.body.operationName) && !user.loggedIn) {
```

Comment:
GraphQL operations except for `login` and `signup` can only be run by authorized users.